Modules/rat_rescue.py

```diff
@@ -784,22 +789,43 @@ def rats(self, value):
                      await RatCache().get_rat_by_name(name))
             if found[0]:
                 self.rats.append(found[0])
-                return
+                assigned_rat = found[0]
             elif found[1]:
                 # a generic match (not platform specific) was found
                 # TODO throw a warning so the invoking method can handle this condition
                 log.warning("A match was found, but it was not the right platform!")
                 self.rats.append(found[1])
-                return
+                assigned_rat = found[1]
 
             else:
                 # lets make a new Rat!
-                if self.rat_board:  # PRAGMA: NOCOVER
-                    raise NotImplementedError  # TODO fetch rat from API
+                # if self.rat_board:  # PRAGMA: NOCOVER
+                #    pass  # TODO fetch rat from API
                 # TODO: fetch rats from API handler, use that data to make a new Rat instance
 
                 rat = Rat(name=name, uuid=guid)
                 self.rats.append(rat)
+                assigned_rat = rat
+
+        if isinstance(guid, UUID):
+            # lets check if we already have this rat in the cache
+            found = await RatCache().get_rat_by_uuid(guid)
+            if found:
+                self.rats.append(found)
+                assigned_rat = found
+            # else:  TODO: placeholder for fetching rats from the API handler
+                # if the specified UUID wasn't found, we need to search the API
+
+        elif isinstance(guid, str):
+            # attempt to coerce into a UUID
+            parsed_guid = UUID(guid)
```

Comment:
code duplicity. i would suggest checking if `guid` is a `str`, and attempt the coercion above line 810, that way you can reuse existing machinery

Reply:
The code above this isn't duplicating since guid can be one of two types, I want to handle them as separate "branches" to make the code easier to follow. That's my personal preference, though.